ssl/t1_lib.c

```diff
@@ -896,9 +896,16 @@ int tls12_check_peer_sigalg(SSL *s, uint16_t sig, EVP_PKEY *pkey)
     /* Should never happen */
     if (pkeyid == -1)
         return -1;
-    /* Only allow PSS for TLS 1.3 */
-    if (SSL_IS_TLS13(s) && pkeyid == EVP_PKEY_RSA)
-        pkeyid = EVP_PKEY_RSA_PSS;
+    if (SSL_IS_TLS13(s)) {
+        /* Disallow DSA for TLS 1.3 */
+        if (pkeyid == EVP_PKEY_DSA) {
+            SSLerr(SSL_F_TLS12_CHECK_PEER_SIGALG, SSL_R_WRONG_SIGNATURE_TYPE);
+            return 0;
+        }
+        /* Only allow PSS for TLS 1.3 */
+        if (pkeyid == EVP_PKEY_RSA)
```

Comment:
No because you can use RSA keys for RSA-PSS. DSA keys can't be used at all for TLS 1.3.